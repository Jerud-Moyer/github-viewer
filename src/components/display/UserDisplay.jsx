import React from 'react';
import { useSelector } from 'react-redux';

const UserDisplay = () => {
  
  const userInfo = useSelector(state => state.userInfo);


  return (
    <div>
      <h1>{userInfo.name}</h1>
      <img src={userInfo.imageUrl}/>
      <h3>followers: {userInfo.followerCount}</h3>
      <h3>following: {userInfo.followingCount}</h3>
      <a href={userInfo.profileLink}>Profile</a>
    </div>
  );
};

export default UserDisplay;
