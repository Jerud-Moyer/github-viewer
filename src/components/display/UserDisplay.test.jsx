import React from 'react';
import { render, cleanup } from '@testing-library/react';
import UserDisplay from './UserDisplay';
import { Provider } from 'react-redux';
import store from '../../store';

describe('UserDisplay component', () => {
  afterEach(() => cleanup());
  it('renders UserDisplay', () => {
    const { asFragment } = render(<Provider store={store}>
      <UserDisplay />
    </Provider>);
    expect(asFragment()).toMatchSnapshot();
  });
});
