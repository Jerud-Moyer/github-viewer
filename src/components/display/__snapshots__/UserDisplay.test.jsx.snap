// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDisplay component renders UserDisplay 1`] = `
<DocumentFragment>
  <div>
    <h1 />
    <img />
    <h3>
      followers: 
    </h3>
    <h3>
      following: 
    </h3>
    <a>
      Profile
    </a>
  </div>
</DocumentFragment>
`;
